AmplifyPlugins/DataStore/Tests/AWSDataStorePluginTests/Subscribe/ObserveQueryTaskRunnerTests.swift

```diff
@@ -64,6 +72,113 @@ class ObserveQueryTaskRunnerTests: XCTestCase {
+    /// - Given: ObserveQuery has started and the first snapshot has been received.
+    /// - When:
+    ///    - modelSyncedEvent is sent to the Hub
+    /// - Then:
+    ///    - ObserveQuery will send a second snapshot
+    ///
+    func testGenerateSnapshotOnObserveQueryWhenModelSynced() async throws {
+        let firstSnapshot = asyncExpectation(description: "first query snapshots")
+        let secondSnapshot = asyncExpectation(description: "second query snapshots")
+        let thirdSnapshot = asyncExpectation(description: "third query snapshot", isInverted: true)
+        let dispatchedModelSyncedEvent = AtomicValue(initialValue: false)
+        let taskRunner = ObserveQueryTaskRunner(
+            modelType: Post.self,
+            modelSchema: Post.schema,
+            predicate: nil,
+            sortInput: nil,
+            storageEngine: storageEngine,
+            dataStorePublisher: dataStorePublisher,
+            dataStoreConfiguration: .default,
+            dispatchedModelSyncedEvent: dispatchedModelSyncedEvent,
+            dataStoreStatePublisher: dataStoreStateSubject.eraseToAnyPublisher())
+        let snapshots = taskRunner.sequence
+        Task {
+            var querySnapshots = [DataStoreQuerySnapshot<Post>]()
+            do {
+                for try await querySnapshot in snapshots {
+                    querySnapshots.append(querySnapshot)
+                    if querySnapshots.count == 1 {
+                        XCTAssertEqual(querySnapshot.items.count, 0)
+                        XCTAssertEqual(querySnapshot.isSynced, false)
+                        await firstSnapshot.fulfill()
+                    } else if querySnapshots.count == 2 {
+                        XCTAssertEqual(querySnapshot.items.count, 0)
+                        XCTAssertEqual(querySnapshot.isSynced, true)
+                        await secondSnapshot.fulfill()
+                    } else if querySnapshots.count == 3 {
+                        XCTFail("Should not receive third snapshot for a Model change")
+                        await thirdSnapshot.fulfill()
+                    }
+                }
+            } catch {
+                XCTFail("Failed with error \(error)")
+            }
+        }
+        
+        await waitForExpectations([firstSnapshot], timeout: 5)
+        
+        dispatchedModelSyncedEvent.set(true)
+        let modelSyncedEventPayload = HubPayload(eventName: HubPayload.EventName.DataStore.modelSynced, data: ModelSyncedEvent(modelName: Post.modelName, isFullSync: true, isDeltaSync: false, added: 0, updated: 0, deleted: 0))
+        Amplify.Hub.dispatch(to: .dataStore, payload: modelSyncedEventPayload)
+        await waitForExpectations([secondSnapshot], timeout: 10)
+        
+        let newModelSyncedEventPayload = HubPayload(eventName: HubPayload.EventName.DataStore.modelSynced, data: ModelSyncedEvent(modelName: Blog6.modelName, isFullSync: true, isDeltaSync: false, added: 0, updated: 0, deleted: 0))
```

Comment:
instead of `newModelSyncedEventPayload`, we can name it something like `modelSyncedEventNotMatch`